docs/GDPs/gdp-0001-standalone-cli.rst

```diff
@@ -0,0 +1,434 @@
+=================================
+GDP 1 — Standalone Compiler CLI
+=================================
+
+:Author: Rico Häuselmann <ricoh@cscs.ch>
+:Status: Draft
+:Type: Feature
+:Created: 17.04.2020
+:Discussion PR: `https://github.com/GridTools/gt4py/pull/21 <discussion_pr>`_
+:Implementation: `https://github.com/GridTools/gt4py/pull/23 <reference_impl_pr>`_
+
+
+Abstract
+--------
+
+GT4Py currently provides a python API / embedded DSL for defining,
+compiling and running GridTools stencils from python programs.
+
+This GDP describes an additional CLI which can be integrated into the build
+process of non-python programs to compile, link and run stencils written
+in GT4Py DSL from any host language that can link to C++ libraries.
+
+A `reference implementation <reference_impl_pr>`_ exists, which will on acception of this GDP be drawn on for adding functionality of this GDP in a series of separate pull requests.
+
+Motivation and Scope
+--------------------
+
+This GDP proposes adding a CLI command named `gtpyc` (naming rationale / alternatives documented somewhere below).
+The command will take a gtscript file and the name of one of the available backends as input and output
+backend specific source code. This includes any language bindings supported by the backend. Commandline options will
+allow full control over what bindings should be created if any, depending on what the backend supports.
+
+A gtscript file denotes a file with a to-be-defined extension (suggestion: `.gt.py`), beginning with the line `## using-dsl: gtscript`. All other content must be valid python under the assumption that the first line has been replaced by `from gt4py.gtscript import *`.
```

Comment:
Thinking a bit more in detail about this, I think it would be better to enforce a more strict format in the extra annotation header with "shebang"-like prefix, so we could extend the annotation syntax in the future if needed with additional keywords. For example, something like `##[GT]` or similar prefixes:
`##[GT] using-dsl:`

Reply:
I think I will go for `# [GT] using-dsl:` as most python linters by default complain about `#` not followed by a space.

Reply:
Fully agree